File: lessons/compiled_lang/go/src/e00.branch.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	stdin := bufio.NewScanner(os.Stdin)
	fmt.Print("Would you like a toast? [Yes/No]: ")
	stdin.Scan()
	response := stdin.Text()

	var message string
	if response == "Yes" {
		message = "That's great!"
	} else {
		message = "How about a muffin?"
	}

	fmt.Println(message)
}

File: lessons/compiled_lang/rust/src/e00.branch.rs
use std::io::{self, Write};

fn main() {
    print!("Would you like a toast? [Yes/No]: ");
    io::stdout().flush().unwrap();

    let mut response = String::new();
    io::stdin().read_line(&mut response).unwrap();
    let response = response.trim_end();

    let message;
    if response == "Yes" {
        message = "That's great!";
    } else {
        message = "How about a muffin?";
    }

    println!("{}", message);
}
